src/sentry/static/sentry/app/utils/withSentryAppComponents.jsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import Reflux from 'reflux';
+import createReactClass from 'create-react-class';
+
+import getDisplayName from 'app/utils/getDisplayName';
+import SentryAppComponentsStore from 'app/stores/sentryAppComponentsStore';
+
+const withSentryAppComponents = (WrappedComponent, {componentType} = {}) =>
+  createReactClass({
+    displayName: `withSentryAppComponents(${getDisplayName(WrappedComponent)})`,
+    mixins: [Reflux.connect(SentryAppComponentsStore, 'components')],
+
+    render() {
+      return (
+        <WrappedComponent
+          components={SentryAppComponentsStore.getComponentByType(componentType)}
```

Comment:
👍 but it probably shouldn't be optional if we're passing like this. Or just do this

```jsx
components={componentType
    ? SentryAppComponentsStore.getComponentByType(componentType)
    : this.state.components}
```

Reply:
@EvanPurkhiser Why is that? The argument is optional for `getComponentByType` in `SentryAppComponentsStore`. It passes the unit test for this case as well btw.

Reply:
Oh, I wouldn’t have expected it to be optional on an api where the method name is explicitly byType. In that case I think the api of the store might be a bit confusing

Reply:
@EvanPurkhiser are you saying I should change the api of the store in this PR?